indexer/src/types/transaction.ts

```diff
@@ -67,28 +99,138 @@ export function toEthTx({
 }
 
 /**
- * @param typeTransaction - Typed transaction to be converted.
- * @param receipt The transaction receipt of the transaction.
+ * Calculates the chain ID from the transaction.
+ *
+ * @param typedTransaction - The typed transaction object.
+ * @param jsonTx - The JSON representation of the transaction.
+ * @returns - The chain ID as a string, or undefined if the chain ID cannot be determined.
+ *
+ * If the transaction is a legacy transaction that supports EIP-155 replay protection,
+ * the chain ID is calculated from the `v` value using the formula:
+ * v = 35 + 2 * chainId + yParity -> chainId = (v - 35) / 2
+ */
+function calculateChainId(typedTransaction: TypedTransaction, jsonTx: JsonTx): string | undefined {
+  if (isLegacyTx(typedTransaction) && typedTransaction.supports(Capability.EIP155ReplayProtection)) {
```

Comment:
you can add

```typescript
  const { v, chainId } = jsonTx;
```

at the beginning of the function to have clear reusable components